File: server/models/domain.js
const { Schema, model } = require("mongoose");

const domainSchema = Schema({
  name: { type: String, required: true, trim: true },
  status: {
    type: String,
    required: true,
    enum: ["active", "inactive"],
  },
  plannedStatus: {
    type: String,
    required: true,
    enum: ["active", "inactive"],
  },
  dtStatus: { type: Date, default: Date.now() },
  dtCr: { type: Date, default: Date.now() },
  dtUp: Date,
});

domainSchema.index({ name: 1 }, { unique: true });

domainSchema.set("toObject", { virtuals: true });
domainSchema.set("toJson", { virtuals: true });

exports.Domain = model("Domain", domainSchema);

File: marina/lib/src/domain/domain/entities/domain.dart
import 'package:equatable/equatable.dart';

class Domain extends Equatable {
  const Domain({
    required this.id,
    required this.name,
    required this.status,
    required this.plannedStatus,
    this.dtStatus,
  });

  const Domain.empty()
    : id = 'Test',
      name = 'empty',
      status = 'inactive',
      plannedStatus = 'inactive',
      dtStatus = null;

  final String id;
  final String name;
  final String status;
  final String plannedStatus;
  final DateTime? dtStatus;

  @override
  List<Object?> get props => [id, name, status, plannedStatus, dtStatus];
}
